src/site/content/en/blog/structured-clone/index.md

```diff
@@ -0,0 +1,121 @@
+- **Functions**: `JSON.stringify()` will quietly discard functions.
+
+## Structured cloning
+
+The platform already needed the ability to create deep copies of JavaScript values in a couple of places: Storing a JS value in IndexedDB requires some form of serialization so it can be stored on disk and later deserialized to restore the JS value. Similarly, sending messages to a WebWorker via `postMessage()` requires transferring a JS value from one JS realm to another. The algorithm that is used for this is called “Structured Clone”, and until recently, wasn’t easily accessible to developers.
+
+That has now changed! The HTML spec was amended to expose a function called `structuredClone()` that runs exactly that algorithm as a means for developers to easily create deep copies of JavaScript values.
+
+```js
+const myDeepCopy = structuredClone(myOriginal);
+```
+
+That’s it! That’s the entire API. If you want to dive deeper into the details, take a look at the [MDN article][mdn structuredclone].
+
+### Features and limitations
+
+Structured cloning addresses many (although not all) shortcomings of the `JSON.stringify()` technique. Structured cloning can handle cyclical data structures, support many built-in data types and is generally more robust and often faster.
+
+However, it still has some limitations that may catch you off-guard:
+ 
+- **Prototypes**: If you use `structuredClone()` with a class instance, you’ll get a plain object as the return
+value, as structured cloning discards the object’s prototype chain. 
+- **Functions**: If your object contains functions, they will be _quiety_ discarded.
+- **Non-cloneables**: Some values are not structured cloneable, most notably `Error` and DOM nodes. It
+will cause `structuredClone()` to throw.
+ 
+If any of these limitations are a deal-breaker for your use-case, libraries like Lodash still provide custom implementations of other deep-cloning algorithms that may or may not fit your use-case.
+
+### Performance
+
+While I haven’t done a new micro-benchmark comparison, [I did a comparison in early 2018][surma blog], before `structuredClone()` was exposed. Back then, `JSON.parse()` was the fastest option for very small objects. I expect that to remain the same. Techniques that relied on structured cloning were (significantly) faster for bigger objects. Considering that the new `structuredClone()` comes without the overhead of abusing other APIs and is more robust  than `JSON.parse()`, I recommend you make it your default approach for creating deep copies.
+ 
+## Conclusion
+
+If you need to create a deep-copy of a value in JS — maybe that be because you use immutable data structures or you want to make sure a function can manipulate an object without affecting the original — you no longer need to reach for workarounds or libraries. The JS ecosystem now has [`structuredClone()`][mdn structuredclone]. Huzzah.
```

Comment:
```suggestion
If you need to create a deep-copy of a value in JS—maybe that be because you use immutable data structures or you want to make sure a function can manipulate an object without affecting the original—you no longer need to reach for workarounds or libraries. The JS ecosystem now has [`structuredClone()`][mdn structuredclone]. Huzzah.
```